dulwich/objects.py

```diff
@@ -1179,7 +1179,12 @@ def lookup_path(self, lookup_obj, path):
           path: Path to lookup
         Returns: A tuple of (mode, SHA) of the resulting path.
         """
-        parts = path.split(b"/")
```

Comment:
Why is this necessary? At the very least, there should be accompanying test changes - but I don't understand what bug in lookup_path this is fixing.

Reply:
this [comment](https://github.com/dulwich/dulwich/pull/898#discussion_r692582168)

Reply:
You're passing a file system path to a function that expects a canonical git path - the caller there should be converting from a filesystem path to a git path and then calling lookup_path().

Reply:
E.g. by calling dulwich.index._fs_to_tree_path,

Reply:
call dulwich.index._fs_to_tree_path now